### PR TITLE
Save proposals sort order to URL

### PR DESCRIPTION
**Pull request by hgwood**
# Behavior before this change

- As an organiser, select an event, then go to "Call for paper" page
- Proposals are listed newest first
- I change the sort order to Highest rating
- I refresh the page
- Proposals are listed newest first

# Behavior after this change

- I change the sort order to Highest rating
- I refresh the page
- Proposals are listed highest rating first

# Detailed behavior

Changing the sort order in the UI updates the route query parameter `sortOrder` to match.

When landing on the `PROPOSALS` route, since there might be a sort order stored in both the router state and the UI state, the app need to reconcile the 2 values. Here's the algorithm that I used:
- Landing without a query parameter `sortOrder` updates the route to the order set in the UI state, unless there is none, in which case the order default to `newest`.
- Landing with a valid* query parameter `sortOrder` updates the UI state to match
- Landing with an invalid* query parameter `sortOrder` results in both the route query parameter and the UI state getting set to `newest`

\* "valid" refers to the value being one of the four values defined in the route config for the route `PROPOSALS`

Summing it up in a table:

Router / UI | No sort order | Has sort order `ui`
----|----|----
No sort order | `newest` | `ui`
Has sort order `router` | `router` | `router`
Has invalid sort order | `newest` | `newest`

### CODE REVIEW

**Comment by hgwood** [comment]
I'm having second thoughts about my implementation. There should not be 2 places in the redux state for the same information, there should be only one. Let me try something else.

**Comment by hgwood** [comment]
I tried to have only the router state carry the sort order, but that had the unforeseen consequence of loosing the sort order when navigating to another route. I don't know if that's a good thing or not but it breaks with the current behavior. What's your opinion?

File: src/store/listeners/reactions/router.js [hunk 1]
```diff
@@ -38,4 +44,39 @@ export const onRouteChanged = reaction((action, store) => {
...
+    const availableSortOrders = getRouterResult(store.getState()).sortOrders
+    const sortOrderFromRouterState = getRouterQuery(store.getState()).sortOrder
+    const sortOrderFromAppState = store.ui.organizer.proposals.get().sortOrder
+    const { shouldUpdateAppState, shouldUpdateRouterState, newValue: sortOrder } = reconcile(
+      sortOrderFromAppState,
+      sortOrderFromRouterState,
+      availableSortOrders,
+    )
+    if (shouldUpdateAppState) {
+      store.dispatch(store.ui.organizer.proposals.update({ sortOrder }))
```
Comment (bpetetot) [inline src/store/listeners/reactions/router.js]: `store.dispatch` is not necessary with k-ramel redux factories. actions are dispatched automatically by calling the action creator.

**Review by bpetetot — CHANGES_REQUESTED** [review]
Very good work ! Thanks 👍 

And yes, I think we should keep also the current behaviour, to keep sorting by navigating and refreshing. It will give a better experience for the user.

File: src/store/listeners/reactions/router.js (hunk 1 above)
Comment (hgwood) [reply src/store/listeners/reactions/router.js]: Right.

Comment (hgwood) [reply src/store/listeners/reactions/router.js]: Fixed.

**Review by bpetetot — APPROVED** [review]

**Pull request merged**